### PR TITLE
fix: added onStepSelected output for progress-indicator component

### PR DESCRIPTION
**Pull request by Licen-it**
Closes IBM/carbon-components-angular#1097

Added `@Output onStepSelected` for progress-indicator component

#### Changelog

**New**

* Progress-indicator component now exposes an output event emitter `onStepSelected` that will emit the step and its index when a click is performed on the step label

### CODE REVIEW

File: src/progress-indicator/progress-indicator.component.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { Component, Input } from "@angular/core";
+import { Component, Input, Output, EventEmitter } from "@angular/core";
```
Comment (Donisius) [inline src/progress-indicator/progress-indicator.component.ts]:
```suggestion
import { 
	Component,
	Input,
	Output,
	EventEmitter
} from "@angular/core";
```

File: src/progress-indicator/progress-indicator.component.ts [hunk 2]
```diff
@@ -64,6 +65,8 @@ export class ProgressIndicator {
 		return steps;
 	}
 
+	@Output() onStepSelected = new EventEmitter<{ step: Step, index: number }>();
```
Comment (Donisius) [inline src/progress-indicator/progress-indicator.component.ts]:
```suggestion
	@Output() stepSelected = new EventEmitter<{ step: Step, index: number }>();
```

Comment (Donisius) [reply src/progress-indicator/progress-indicator.component.ts]: Just to be consistent with the naming patterns for selected outputs

File: src/progress-indicator/progress-indicator.component.spec.ts [hunk 3]
```diff
@@ -7,9 +7,14 @@ import { CommonModule } from "@angular/common";
...
 import { CheckmarkOutline16Module } from "@carbon/icons-angular/lib/checkmark--outline/16";
 import { Warning16Module } from "@carbon/icons-angular/lib/warning/16";
+import { Step } from "./progress-indicator-step.interface";
 
 @Component({
-	template: `<ibm-progress-indicator [steps]="steps" [current]="current"></ibm-progress-indicator>`
+	template: `<ibm-progress-indicator
+					[steps]="steps"
+					[current]="current"
+					(onStepSelected)="onStepSelected.emit($event)">
```
Comment (Donisius) [inline src/progress-indicator/progress-indicator.component.spec.ts]:
```suggestion
					(stepSelected)="onStepSelected.emit($event)">
```

Comment (Licen-it) [reply src/progress-indicator/progress-indicator.component.spec.ts]: done...i also added the action in the story

**Pull request merged**